upcloud/managed_database.go

```diff
@@ -0,0 +1,693 @@
+	ManagedDatabaseServiceTypeMysql = "mysql"
+)
+
+// Metric periods
+const (
+	// ManagedDatabaseMetricPeriodHour represents the observation period of metrics request
+	ManagedDatabaseMetricPeriodHour = "hour"
+	// ManagedDatabaseMetricPeriodDay represents the observation period of metrics request
+	ManagedDatabaseMetricPeriodDay = "day"
+	// ManagedDatabaseMetricPeriodWeek represents the observation period of metrics request
+	ManagedDatabaseMetricPeriodWeek = "week"
+	// ManagedDatabaseMetricPeriodMonth represents the observation period of metrics request
+	ManagedDatabaseMetricPeriodMonth = "month"
+	// ManagedDatabaseMetricPeriodYear represents the observation period of metrics request
+	ManagedDatabaseMetricPeriodYear = "year"
+)
+
+// Log query constants
+const (
+	// ManagedDatabaseLogOrderAscending can be used to query logs in ascending order
+	ManagedDatabaseLogOrderAscending = "asc"
+	// ManagedDatabaseLogOrderDescending can be used to query logs in descending order
+	ManagedDatabaseLogOrderDescending = "desc"
+)
+
+// Common Managed Database properties
+const (
+	// ManagedDatabasePropertyAutoUtilityIPFilter enables automatic ip filter generation from utility network
+	// within the same zone.
+	ManagedDatabasePropertyAutoUtilityIPFilter = "automatic_utility_network_ip_filter"
+	// ManagedDatabasePropertyIPFilter allows adjusting the custom IP filter of a service. The value should
+	// contain a slice of strings representing individual IP addresses or IP addresses with CIDR mask.
+	// Currently IPv4 addresses or networks are supported.
+	ManagedDatabasePropertyIPFilter = "ip_filter"
+	// ManagedDatabaseAllIPv4 property value can be used together with ManagedDatabasePropertyIPFilter to allow access from all
+	// IPv4 hosts.
+	ManagedDatabaseAllIPv4 = "0.0.0.0/0"
+	// ManagedDatabasePropertyPublicAccess enables public access via internet to the service. A separate public
+	// endpoint DNS name will be available under Components after enabling.
+	ManagedDatabasePropertyPublicAccess = "public_access"
+)
+
+// User types
+const (
+	// ManagedDatabaseUserTypePrimary is a type of the primary user of a managed database service. There can be only
+	// one primary user per service. The primary user has administrative privileges to manage logical databases and
+	// users thru the database's native API.
+	ManagedDatabaseUserTypePrimary = "primary"
+	// ManagedDatabaseUserTypeNormal is a type of a normal database user of a managed database service. There can
+	// be multiple normal users and the primary user can manage the privileges of these users thru the database's
+	// native API.
+	ManagedDatabaseUserTypeNormal = "normal"
+)
+
+// Authentication types
+const (
+	// ManagedDatabaseUserAuthenticationCachingSHA2Password selects "caching_sha2_password" type of authentication type.
+	// This type is only supported with MySQL services.
+	//nolint:gosec
+	ManagedDatabaseUserAuthenticationCachingSHA2Password = "caching_sha2_password"
```

Comment:
I'm guessing the gosec warning is related to it thinking this is a password due to the name? It would be good to add a nolint comment clarifying it, eg something like `//nolint:gosec // this is not actually a password but authentication type`

Reply:
fair point, will add.

Reply:
implemented